src/core/client/framework/lib/storage/PymStorage.ts

```diff
@@ -0,0 +1,71 @@
+import { Child, Parent } from "pym.js";
+import uuid from "uuid/v4";
+
+export interface PymStorage {
+  /**
+   * value = storage[key]
+   */
+  getItem(key: string): Promise<string | null>;
+  /**
+   * delete storage[key]
+   */
+  removeItem(key: string): Promise<void>;
+  /**
+   * storage[key] = value
+   */
+  setItem(key: string, value: string): Promise<void>;
+}
+
+type Pym = Child | Parent;
+
+/**
+ * Creates a storage that put requests onto pym.
+ * This is the counterpart of `connectStorageToPym`.
+ * @param  {string} pym  pym
+ * @return {Object} storage
+ */
+export default function createPymStorage(
+  pym: Pym,
+  type: "localStorage" | "sessionStorage"
+): PymStorage {
```

Comment:
Isn't this best represented as another class that implements PymStorage?

Reply:
Copied that from v4. I could turn it into a class, but is it really going to be "better"? 🤔